File: Ahorn/entities/FlameCerberus.jl
module FlameCerberus

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/FlameCerberus" FlameCerberus(x::Integer, y::Integer, detectionRange::Integer=200, headHealth::Integer=3, moveSpeed::Integer=60)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(FlameCerberus)
)

function Ahorn.selection(entity::FlameCerberus)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::FlameCerberus, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/flameCerberus.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local flameCerberus = {}

flameCerberus.name = "MaggyHelper/FlameCerberus"
flameCerberus.depth = -100
flameCerberus.fieldInformation = {
    headHealth = {
        minimumValue = 1
    },
    detectionRange = {
        minimumValue = 0
    },
    moveSpeed = {
        minimumValue = 0
    }
}
flameCerberus.placements = {
    {
        name = "normal",
        data = {
            headHealth = 3,
            detectionRange = 200,
            moveSpeed = 60
        }
    }
}

function flameCerberus.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/flame_cerberus/idle", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function flameCerberus.selection(room, entity)
    return utils.rectangle(entity.x - 40, entity.y - 48, 80, 48)
end

return flameCerberus
